Sourcery/main.swift

```diff
@@ -67,8 +67,22 @@ fileprivate enum Validators {
         return path
     }
 
-    static func isWriteable(path: Path) -> Path {
-        if path.exists && !path.isWritable {
+    static func isWritable(path: Path, autoLock: Bool) -> Path {
+        let pathExistsAndIsNotWritable = path.exists && !path.isWritable
+        let pathExistsAndWillNotBeWritableIfUnlocked: Bool = {
+            guard path.exists, autoLock, path.isImmutable else {
+                return false
+            }
+            
+            try! path.setImmutable(false)
```

Comment:
I got the idea, but if the path can be unlocked by the Sourcery it does not mean it's safe to do that. Is there any way to set owner of the file/dir and check if it's Sourcery? 
Also I don't think it's safe to use `try!` here. It can easily fail and we should not crash in this situation. It's probably better to wrap it with do-catch:

```
do {
  try path.setImmutable(false)
  let isWritableIfUnlocked = path.isWritable
  try path.setImmutable(true)
} catch {
...
}
```